xlators/features/locks/src/posix.c

```diff
@@ -2041,21 +2039,16 @@ do_blocked_rw(pl_inode_t *pl_inode)
 
     INIT_LIST_HEAD(&wind_list);
 
-    pthread_mutex_lock(&pl_inode->mutex);
+    list_for_each_entry_safe(rw, tmp, &pl_inode->rw_list, list)
     {
-        list_for_each_entry_safe(rw, tmp, &pl_inode->rw_list, list)
-        {
-            if (__rw_allowable(pl_inode, &rw->region, rw->stub->fop)) {
-                list_del_init(&rw->list);
-                list_add_tail(&rw->list, &wind_list);
-                if (pl_inode->mlock_enforced &&
-                    pl_inode->track_fop_wind_count) {
-                    pl_inode->fop_wind_count++;
-                }
+        if (__rw_allowable(pl_inode, &rw->region, rw->stub->fop)) {
+            list_del_init(&rw->list);
+            list_add_tail(&rw->list, &wind_list);
+            if (pl_inode->mlock_enforced && pl_inode->track_fop_wind_count) {
+                pl_inode->fop_wind_count++;
             }
         }
     }
-    pthread_mutex_unlock(&pl_inode->mutex);
 
     list_for_each_entry_safe(rw, tmp, &wind_list, list)
```

Comment:
This loop cannot be executed inside a loop. It can cause lock contention and deadlocks.